### ISSUE TITLE
TypeError when running `node.dht.findProvs` from js-ipfs

### ISSUE THREAD

**Issue by jawerty** [libp2p/js-libp2p-kad-dht#112]
Hey I'm having a pretty simple issue where if I try to find providers (or provide) using the ipfs module since there are no peers providing the hash i'm requesting it should simply not return anything but it fails on an error log in `libp2p-kad-dht/src/query/index.js:100` since thus.run.errors and this.run.peersSeen is never set at that point. I patched it locally but thought you may want to update the to catch this error when the Run() constructor is never called.

```
TypeError: Cannot read property 'length' of undefined
    at Query.stop (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/query/index.js:100:34)
    at timeout (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/private.js:527:15)
    at /Users/jared/Documents/battalion/node_modules/async/timeout.js:79:26
    at Query.run (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/query/index.js:64:14)
    at timeout (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/private.js:526:29)
    at /Users/jared/Documents/battalion/node_modules/async/timeout.js:86:12
    at /Users/jared/Documents/battalion/node_modules/async/internal/initialParams.js:11:12
    at dht.providers.getProviders (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/private.js:526:61)
    at _getProvidersMap (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/providers.js:268:7)
    at Providers._getProvidersMap (/Users/jared/Documents/battalion/node_modules/libp2p-kad-dht/src/providers.js:172:5)
```

**Comment by jacobheun** [comment]
Thanks for submitting the issue! It will be fixed by https://github.com/libp2p/js-libp2p-kad-dht/pull/116

**Comment by kumavis** [comment]
Fixed by #116
